src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,122 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    public static final int DEFAULT_CAPACITY = 10;
+    public static final double GROWTH_MULTIPLIER = 1.5;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        growCapacityIfArrayFull();
+        elements[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
+        validateIndexForAddition(index);
+        if (size + 1 >= elements.length) {
+            growCapacity();
+        }
+        System.arraycopy(elements, index, elements, index + 1, size - index);
+        elements[index] = value;
+        size++;
 
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        while (elements.length < (size + list.size())) {
+            growCapacity();
+        }
+        for (int i = 0; i < list.size(); i++) {
+            elements[size + i] = list.get(i);
+        }
+        size += list.size();
```

Comment:
not critical, but think about how to reuse add method here to make code more simple